File: sic-api/Entities/Su/SuProfile.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using sic_api.Attributes;
using sic_api.Entities;
using sic_api.Entities.Db;
using sic_api.Model.Storage;

namespace sic_api.Entities.Su;


[Index(nameof(UserId), IsUnique = true)]
[Table("su_profile")]
public class SuProfile : BaseEntity
{
    [Required]
    [MaxLength(100)]
    [Column("user_id")]
    public string UserId { get; set; } = default!;

    [MaxLength(30)]
    [Column("tax_id")]
    public string? TaxId { get; set; }

    [Required]
    [ForeignKey(nameof(Title))]
    [Column("title_id")]
    public Guid TitleId { get; set; }
    public DbTitle Title { get; set; } = default!;

    [Required]
    [MaxLength(100)]
    [Column("first_name_en")]
    public string FirstNameEn { get; set; } = default!;

    [MaxLength(100)]
    [Column("middle_name_en")]
    public string? MiddleNameEn { get; set; }

    [MaxLength(100)]
    [Column("last_name_en")]
    public string? LastNameEn { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("first_name_local")]
    public string FirstNameLocal { get; set; } = default!;

    [MaxLength(100)]
    [Column("middle_name_local")]
    public string? MiddleNameLocal { get; set; }

    [MaxLength(100)]
    [Column("last_name_local")]
    public string? LastNameLocal { get; set; }

    [ForeignKey(nameof(Country))]
    [Column("country_id")]
    public Guid? CountryId { get; set; }
    public DbCountry? Country { get; set; }

    [Column("support_local_address")]
    public bool SupportLocalAddress { get; set; } = false;
    
    [MaxLength(255)]
    [Column("address_en")]
    public string? AddressEn { get; set; }

    [MaxLength(255)]
    [Column("address_local")]
    public string? AddressLocal { get; set; }

    [ForeignKey(nameof(Province))]
    [Column("province_id")]
    public Guid? ProvinceId { get; set; }
    public DbProvince? Province { get; set; }

    [ForeignKey(nameof(District))]
    [Column("district_id")]
    public Guid? DistrictId { get; set; }
    public DbDistrict? District { get; set; }

    [ForeignKey(nameof(SubDistrict))]
    [Column("sub_district_id")]
    public Guid? SubDistrictId { get; set; }
    public DbSubDistrict? SubDistrict { get; set; }

    [MaxLength(20)]
    [Column("zip_code")]
    public string? ZipCode { get; set; }

    [Required]
    [MaxLength(320)]
    [Column("email")]
    public string Email { get; set; } = default!;
    
    [MaxLength(20)]
    [Column("phone_number")]
    public string? PhoneNumber { get; set; }

    [Column("upload_group_id")]
    public Guid? UploadGroupId { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuProfile.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import com.softinter.sicapi.entity.db.*;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "su_profile",
       indexes = {
           @Index(name = "idx_user_id", columnList = "user_id", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuProfile extends BaseEntity {

    @Column(name = "user_id", nullable = false, length = 100)
    private String userId;

    @Column(name = "tax_id", length = 30)
    private String taxId;

    // Foreign key fields (mirror, read-only)
    @Column(name = "title_id", nullable = false, insertable = false, updatable = false)
    private UUID titleId;

    @Column(name = "country_id", insertable = false, updatable = false)
    private UUID countryId;

    @Column(name = "province_id", insertable = false, updatable = false)
    private UUID provinceId;

    @Column(name = "district_id", insertable = false, updatable = false)
    private UUID districtId;

    @Column(name = "sub_district_id", insertable = false, updatable = false)
    private UUID subDistrictId;

    // Navigation properties (own the foreign key columns)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "title_id", nullable = false)
    private DbTitle title;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "country_id")
    private DbCountry country;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "province_id")
    private DbProvince province;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "district_id")
    private DbDistrict district;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sub_district_id")
    private DbSubDistrict subDistrict;

    // Other fields
    @Column(name = "first_name_en", nullable = false, length = 100)
    private String firstNameEn;

    @Column(name = "middle_name_en", length = 100)
    private String middleNameEn;

    @Column(name = "last_name_en", length = 100)
    private String lastNameEn;

    @Column(name = "first_name_local", nullable = false, length = 100)
    private String firstNameLocal;

    @Column(name = "middle_name_local", length = 100)
    private String middleNameLocal;

    @Column(name = "last_name_local", length = 100)
    private String lastNameLocal;

    @Column(name = "support_local_address", nullable = false)
    private Boolean supportLocalAddress = false;

    @Column(name = "address_en", length = 255)
    private String addressEn;

    @Column(name = "address_local", length = 255)
    private String addressLocal;

    @Column(name = "zip_code", length = 20)
    private String zipCode;

    @Column(name = "email", nullable = false, length = 320)
    private String email;

    @Column(name = "phone_number", length = 20)
    private String phoneNumber;

    @Column(name = "upload_group_id")
    private UUID uploadGroupId;

    @Column(name = "is_active")
    private Boolean isActive; 
}